ganga/GangaCore/Utility/Monitoring.py

```diff
@@ -0,0 +1,54 @@
+from GangaCore.Utility.Config import getConfig
+
+import os
+from memory_profiler import profile
+import cProfile
+import time
+
+timestr = time.strftime("-%Y%m%d-%H%M%S")
+
+c = getConfig('Configuration')
+
+path = os.path.join(c['gangadir'],'logs')
+
+def makedir(path):
+    try:
+        if not os.path.exists(path):
+            os.makedirs(path)
+        return path
+    except Exception as e:
+        pass
+    # Will do error handling later
+
+makedir(path)
+
+def cpu_profile(func):
+    mpath = makedir(os.path.join(path,'cpu_profile/'))
+    def wrapper(*args, **kwargs):
+        datafn = mpath+func.__name__ +timestr+".profile"
+        prof = cProfile.Profile()
+        retval = prof.runcall(func, *args, **kwargs)
+        prof.dump_stats(datafn)
+        return retval
+    return wrapper
+
+def cpu_profiler(cls, profile_cpu=c['Profile_CPU']):
+    if profile_cpu:
+        for key, value in list(vars(cls).items()):
```

Comment:
`items()` presumably already returns a list so the explicit list construction seems irrelevant. Also since these are temporary intermediary lists simply used in the iteration perhaps a generator would be better.

i.e. does this not simplify to:
```python
for key, value in vars(cls).iteritems():
```

Reply:
Yes. Thank you for this.